src/lib.rs

```diff
@@ -0,0 +1,343 @@
+//! Rate limiting for Rama applications using the governor crate
+//!
+//! This crate provides a `GovernorPolicy` that can be used with Rama's `LimitLayer`
+//! for rate limiting HTTP requests or any other kind of request.
+
+use std::collections::HashSet;
+use std::fmt;
+use std::num::NonZeroU32;
+use std::sync::{Arc, Mutex};
+use std::time::Duration;
+
+use governor::{DefaultDirectRateLimiter, DefaultKeyedRateLimiter, Quota};
+use once_cell::sync::{Lazy, OnceCell};
+use rama::Context;
```

Comment:
I would suggest to depend directly on `rama_core` and `rama_http` for your crate. That's the only dependencies you truly need. `rama` crate is meant for end-users. You can of course still use `rama` as a `dev-dependency`